lib/shopify_cli/environment.rb

```diff
@@ -87,6 +92,20 @@ def self.partners_domain(env_variables: ENV)
       end
     end
 
+    def self.spin_url_override(env_variables: ENV)
+      tokens = SPIN_OVERRIDE_ENV_NAMES.map do |name|
+        env_variables[name]
+      end
+
+      return if tokens.all? { |token| token.nil? }
+
+      if tokens.any? { |token| token.nil? }
+        raise "To manually target a spin instance, you must set #{SPIN_OVERRIDE_ENV_NAMES}"
```

Comment:
If you set the workspace, namespace, and host, we'll use that instead of asking the spin tool.

This is useful if you ever need to interact with someone else's instance.